chain/network/src/tests/util.rs

```diff
@@ -0,0 +1,82 @@
+#![allow(dead_code)]
+use once_cell::sync::Lazy;
+
+// TODO: consider wrapping these types to prevent interaction with
+// real time in tests.
+pub type Instant = std::time::Instant;
+pub type Utc = chrono::DateTime<chrono::Utc>;
+pub type Duration = chrono::Duration;
+
+// Instant doesn't have a deterministic contructor,
+// however since Instant is not convertible to an unix timestamp,
+// we can snapshot Instant::now() at the process startup
+// and treat it as a constant. All observable effects will be then
+// deterministic.
+static FAKE_CLOCK_MONO_START: Lazy<Instant> = Lazy::new(Instant::now);
+
+// An arbitrary non-trivial deterministic Utc timestamp.
+static FAKE_CLOCK_UTC_START: Lazy<Utc> =
+    Lazy::new(|| Utc::from(std::time::SystemTime::UNIX_EPOCH) + Duration::seconds(89108233));
+
+enum ClockInner<'a> {
+    Real,
+    Fake(&'a FakeClock),
+}
+
+pub struct Clock<'a>(ClockInner<'a>);
```

Comment:
add a comment explaining when we should use this Clock vs the MockClock that we have in primitives/src/time.rs

Reply:
done